### PR TITLE
Get development working!

### PR DESCRIPTION
**Pull request by TomFanella4**
1. Many small bug fixes
1. Removed unused code
1. The Client now tracks the user's location

### CODE REVIEW

File: backend/Handlers.go [hunk 1]
```diff
@@ -48,15 +54,23 @@ const (
...
+			return
+		}
+		var profile UserProfileResponse
+		profile.Profile = user.Profile
+		profile.Location = user.Location
+		json.NewEncoder(w).Encode(profile)
+		return
 	}
-	json.NewEncoder(w).Encode(profile)
+	respondWithError(w, Unauthorized, "Unauthorized")
```
Comment (mpace965) [inline backend/Handlers.go]: We talked offline, but `CheckAuthorized` automatically does a `returnWithError` if the user is unauthorized.

File: backend/Storage.go [hunk 2]
```diff
@@ -25,11 +25,24 @@ var fbClient *firego.Firebase
 const separator = "|"
 
 // GetProspectiveUsers Get the list of people for matching in the area
-func GetProspectiveUsers(coords Geolocation, radius int, lastUpdate int) ([]User, error) {
+func GetProspectiveUsers(coords *Geolocation, radius int, lastUpdate int) ([]User, error) {
```
Comment (mpace965) [inline backend/Storage.go:28]: Why does this need to be a pointer?

File: backend/Handlers.go [hunk 3]
```diff
@@ -164,25 +154,21 @@ func Match(w http.ResponseWriter, r *http.Request) {
...
-			return // no match was returned
+		user, err := GetUser(userID)
+		if err != nil {
+			fmt.Println(err)
+			return // Unable to get location
 		}
 		radius := 1     //1 km
 		lastUpdate := 2 // 2hrs
-		PMatchList, err := GetProspectiveUsers(userLocation, radius, lastUpdate)
+		PMatchList, err := GetProspectiveUsers(&user.Location, radius, lastUpdate)
```
Comment (mpace965) [inline backend/Handlers.go:165]: If you remove the pointer from the declaration, be sure to remove the `&` here

**Review by mpace965 — APPROVED** [review]
Looks good, few minor changes

File: backend/Storage.go (hunk 2 above)
Comment (TomFanella4) [reply backend/Storage.go:28]: Will be addressed in a future update

File: backend/Handlers.go (hunk 3 above)
Comment (TomFanella4) [reply backend/Handlers.go:165]: Will be addressed in a future update

File: backend/Handlers.go (hunk 1 above)
Comment (TomFanella4) [reply backend/Handlers.go]: Done

File: backend/Handlers.go [hunk 4]
```diff
@@ -252,8 +235,8 @@ func InitiateMeetover(w http.ResponseWriter, r *http.Request) {
 			fmt.Println(err.Error())
 			return
 		}
+		json.NewEncoder(w).Encode("Success")
```
Comment (raokrutarth) [inline backend/Handlers.go]: This is not a valid JSON response. Try to avoid this style of response

Comment (TomFanella4) [reply backend/Handlers.go]: For sure, this was put in the morning before the demo and should be changed.

**Pull request merged**